orca-mine/src/main/groovy/com/netflix/spinnaker/orca/mine/tasks/CleanupCanaryTask.groovy

```diff
@@ -36,10 +37,43 @@ class CleanupCanaryTask extends AbstractCloudProviderAwareTask implements Task {
 
   @Override
   TaskResult execute(Stage stage) {
+    def canary = stage.mapTo("/canary", Canary)
+    if (canary.health?.health == Health.UNHEALTHY.health && !canary.canaryConfig?.actionsForUnhealthyCanary?.find {
+      it.action == Action.TERMINATE
+    }) {
+      // should succeed (and not perform any cleanup) when the canary is unhealthy and not configured to terminate
+      return DefaultTaskResult.SUCCEEDED
```

Comment:
In the event that `TERMINATE` is not an unhealthy canary action, the `CleanupCanaryTask` should immediately succeed and not destroy the canary clusters.